### ISSUE TITLE
performance regression since Pods 2.3.10

### ISSUE THREAD

**Issue by hotzeplotz** [pods-framework/pods#2145]
as discussed recently on IRC, a major performance regression hit one of my sites running Pods after upgrading from Pods 2.3.9 to 2.3.10, with regression consistently happening throughout to Pods 2.4.

as i'm still investigating whether it is my theme's code which does something it's not supposed to do or if my code is triggering some rare regression within Pods itself, i'm posting here all the info i have so far (it may well be an issue in my code, and in this case it may still be useful to emit a warning within Pods when a suboptimal code path is used in themes)

this site is using 38 Pods with 514 fields in total (a bit less at the time the issue first occurred when Pods 2.3.10 was released) and started life as a Pods 1.x site, upgraded to 2.x along the way.

the issue is a massive slowdown which occurs at every request: uncached page generation times went from 2-3 seconds to over 90 seconds (so much so that the frontend Varnish cache normally times out at its default 60 seconds setting, still waiting for the first byte from the backend Apache2).

i run git bisect between 2.3.9 and 2.3.10 and eventually identified commit pods-framework/pods@e64c9b3 as the one which introduced changes that trigger the slowdown in my theme.

still as of 2.4, everything runs smoothly if i rebase 2.4 to the commit prior to pods-framework/pods@e64c9b3, simply removing the 'bad' commit.

in more detail, what happens within the PHP engine while a slow page generation occurs can be seen by attaching to the affected apache2 process via `sudo strace -p <pid>`:

most of the function calls (several dozens per second) are:

```
lstat("/srv/web/wordpress/www/htdocs/wp-content/plugins/pods/classes/PodsView.php", {st_mode=S_IFREG|0644, st_size=17516, ...}) = 0
```

this happens with W3TC caching enabled (either memcached or XCode); if i disable W3TC and enable pods-alternative-cache instead, there still are huge numbers of repeated lstat() calls, only to the plugin's own disk-based cache objects:

```
access("/srv/web/wordpress/www/htdocs/wp-content/podscache/7-d426a25a698b48b761549551c10aeae8.php", R_OK) = 0
lstat("/srv/web/wordpress/www/htdocs/wp-content/podscache/7-d426a25a698b48b761549551c10aeae8.php", {st_mode=S_IFREG|0644, st_size=7399, ...}) = 0
open("/srv/web/wordpress/www/htdocs/wp-content/podscache/7-d426a25a698b48b761549551c10aeae8.php", O_RDONLY) = 18
```

interestingly, the slowdown is not only triggered when a frontend page is generated, but also while using the WP backend - my theme has a couple of calls to `pods_ui_manage()` which further increase page loads even when just accessing _any_ page on the frontend (i guess this is because `pods_ui_manage()` is called while the WP side menu is generated).

**Comment by hotzeplotz** [comment]
fantastic, 2.5 works great for us and fixes the regression of 2.3.10 - thanks @jamesgol and everyone who contributed to this!

**Comment by sc0ttkclark** [comment]
Awesome!
